pkg/handlers/generic/lifecycle/handlers.go

```diff
@@ -102,17 +104,101 @@ func (h *Handlers) AllHandlers(mgr manager.Manager) []handlers.Named {
 			helmChartInfoGetter,
 		),
 	}
-	return []handlers.Named{
+	allHandlers := []handlers.Named{
 		calico.New(mgr.GetClient(), h.calicoCNIConfig, helmChartInfoGetter),
 		cilium.New(mgr.GetClient(), h.ciliumCNIConfig, helmChartInfoGetter),
+		ccm.New(mgr.GetClient(), ccmHandlers),
 		nfd.New(mgr.GetClient(), h.nfdConfig, helmChartInfoGetter),
 		clusterautoscaler.New(mgr.GetClient(), h.clusterAutoscalerConfig, helmChartInfoGetter),
-		servicelbgc.New(mgr.GetClient()),
 		csi.New(mgr.GetClient(), csiHandlers),
-		ccm.New(mgr.GetClient(), ccmHandlers),
-		serviceloadbalancer.New(mgr.GetClient(), serviceLoadBalancerHandlers),
 		snapshotcontroller.New(mgr.GetClient(), h.snapshotControllerConfig, helmChartInfoGetter),
+		servicelbgc.New(mgr.GetClient()),
+		// The order of the handlers in the list is important and are called consecutively.
+		// The MetalLB provider may be configured to create a IPAddressPool on the remote cluster.
+		// However, the MetalLB provider also has a webhook that validates IPAddressPool requests.
+		// Because this webhook relies on CNI and CCM to already be installed on the remote cluster,
+		// we are placing this handler after the CNI and CCM handlers.
+		serviceloadbalancer.New(mgr.GetClient(), serviceLoadBalancerHandlers),
```

Comment:
I pushed a commit to reorder the handlers with the `serviceloadbalancer` going last, and moved CCM higher up since the Nodes won't become ready until CCM runs.

If we don't want this change here I'm happy to remove this commit

Reply:
This is a good change! Once we change this to run all handlers and aggregate even even failed responses then this won't be a problem.

Reply:
I think it's a good change, but since this PR didn't re-order handlers to begin with, I think it would be better in a separate PR. I appreciate the explanatory comment, and I can add more of them.